src/odemis/acq/fastem.py

```diff
@@ -349,16 +372,16 @@ def run(self):
         # Update the position of the first tile.
         self._pos_first_tile = self.get_pos_first_tile()
 
-        if self._pre_calibrate:
+        if self._pre_calibrations:
             # The pre-calibrations should run on a position that lies a full field
             # outside the ROA, therefore temporarily set the overlap to zero.
             overlap_init = self._roa.overlap
             self._roa.overlap = 0
             # Move the stage to the tile with index (-1, -1), to ensure the autofocus and image translation pre-align
             # are done to the top left of the first field, outside the region of acquisition to limit beam damage.
             self.field_idx = (-1, -1)
-            self.move_stage_to_next_tile()
-            self.pre_calibrate()
+            stage_pos_x, stage_pos_y = self.get_abs_stage_movement()
```

Comment:
you could also move this to inside the pre_calibrate function. If I then update it to have the image_translation_prealign at the center of the megafield it will be easier to add. My code in the pre_calibrate function will then be something like this:

```
    self.field_idx = (-1, -1)
    stage_pos = self.get_abs_stage_movement()
    f = align(self._scanner, self._multibeam,
              self._descanner, self._detector,
              self._stage, self._ccd,
              self._beamshift, self._det_rotator,
              calibrations=[Calibrations.OPTICAL_AUTOFOCUS], 
              stage_pos=stage_pos)


    f.result()  # wait for the calibrations to be finished

    self.field_idx = ....... # half of the max of the megafield
    stage_pos = self.get_abs_stage_movement()

    f = align(self._scanner, self._multibeam,
              self._descanner, self._detector,
              self._stage, self._ccd,
              self._beamshift, self._det_rotator,
              calibrations=[Calibrations.IMAGE_TRANSLATION_PREALIGN], 
              stage_pos=stage_pos)

    f.result()  # wait for the calibrations to be finished

```

Reply:
With "this" you refer to only the stage movement right? I had thought about it before as well. For the field index and overlap I thought it will be easier to leave it outside the pre calibration function to provide more flexibility. But that you had also in mind I guess?

Reply:
We would need to define `self._pos_first_tile` in the test case then. Otherwise, the test on the pre-calibration fails as this attribute is still None.

Reply:
> With "this" you refer to only the stage movement right? I had thought about it before as well. For the field index and overlap I thought it will be easier to leave it outside the pre calibration function to provide more flexibility. But that you had also in mind I guess?

Yes, that's what I had in mind.

> We would need to define self._pos_first_tile in the test case then. Otherwise, the test on the pre-calibration fails as this attribute is still None. What do you think?

I think we already define the `self._pos_first_tile` in a few test cases.